include/dali/core/cuda_utils.h

```diff
@@ -21,20 +21,18 @@
 #include "dali/core/host_dev.h"
 #include "dali/core/dynlink_cuda.h"
 #include "dali/core/cuda_error.h"
-
-// For the CPU we use half_float lib and float16_cpu type
-namespace half_float {
-
-class half;
-
-}
+#ifndef __CUDA_ARCH__
+#include "dali/util/half.hpp"
+#endif
 
 namespace dali {
 
 // For the GPU
-typedef __half float16;
-// For the CPU
-typedef half_float::half float16_cpu;
+#ifdef __CUDA_ARCH__
+using float16 = __half;
+#else
+using float16 = half_float::half;
+#endif
```

Comment:
I am missing the changes in old type switches?

Reply:
I changed them - but now we don't need separate switch for _CPU and _GPU - I'll unify them.

Reply:
Ok, then let's make a follow PR where we move this to core and remove the old switches (_CPU, _GPU)